src/resources/SearchAnalysis/SearchAnalysisInterfaces.ts

```diff
@@ -0,0 +1,143 @@
+import {PostSearchBodyQueryParams} from '../Search/SearchInterfaces';
+
+type RequestParameters = PostSearchBodyQueryParams;
+interface Result {
+    title: string;
+    rankingInfo: unknown;
+    fields: Record<string, string>;
+}
+
+interface ExecutionStep {
+    type: string;
+    name: string;
+}
+
+interface Condition {
+    id: string;
+}
+
+interface AffectedRequestParameters {
+    affectedRequestParameters: Array<{name: string; value: unknown}>;
+}
+
+interface AffectedResultsPosition {
+    affectedResultsPosition: number[];
+}
+
+interface AppliedRules<Affected = Record<string, unknown>, OtherPropertiesApplied = Record<string, unknown>> {
+    appliedCount: number;
+    applied: Array<
+        {
+            id: string;
+            condition?: Condition;
+        } & Affected &
+            OtherPropertiesApplied
+    >;
+}
+
+interface QueryPipelineSelection extends ExecutionStep {
+    type: 'pipelineSelection';
+    selectedPipeline: {
+        id: string;
+        name: string;
+        condition?: Condition;
+    };
+    selectionCause: {
+        type:
+            | 'AUTHENTICATION_PIPELINE'
+            | 'AUTHENTICATION_SEARCH_HUB'
+            | 'ML_ROUTING'
+            | 'PARAMETER_PIPELINE'
+            | 'CONDITION_ROUTING'
+            | 'DEFAULT';
```

Comment:
That's a mouthful.

- [ ] Verify there are no existing or similar type or enum in the code, if so, DRY and refactor.
- [ ] If not, refactor either into a separated type or enum.